Tests/SentryTests/SentryClientTests.swift

```diff
@@ -445,8 +449,11 @@ class SentryClientTest: XCTestCase {
     }
 
     func testCaptureErrorWithSession() {
-        let eventId = fixture.getSut().captureError(error, with: fixture.session, with: Scope())
-        
+        let eventId = fixture.getSut().captureError(error, with: Scope()) { increaseErrorCount in
+            XCTAssertTrue(increaseErrorCount)
+            return self.fixture.session
+        }
```

Comment:
`m`: To guarantee that this block gets called, we should use a `XCTestExpectation`. Same applies for the block below.
```suggestion
        let sessionBlockExpectation = expectation(description: "session block gets called")
        let eventId = fixture.getSut().captureError(error, with: Scope()) { increaseErrorCount in
            XCTAssertTrue(increaseErrorCount)
            sessionBlockExpectation.fulfill()
            return self.fixture.session
        }
        wait(for: [sessionBlockExpectation], timeout: 0.2)
```